crates/core/tedge_api/src/serialize.rs

```diff
@@ -42,6 +42,9 @@ pub enum MeasurementStreamError {
 
     #[error("Unexpected start of group")]
     UnexpectedStartOfGroup,
+
+    #[error("Unexpected type within a group")]
+    UnexpectedType,
```

Comment:
To be removed as the type is not a tedge concept but a c8y one.

Reply:
fixed